src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -416,9 +419,27 @@ public boolean isHalted() {
 
     /**
      * This method orders the {@link TickerTask} to be halted.
+     * Once halted the task <strong>cannot</strong> be un-halted.
+     * This method will block until all sync tasks are processed; however,
+     * it will not block for the async tasks
      */
     public synchronized void halt() {
+        /*
+         * Mark the ticker as halted so that #tick won't attempt to schedule another task
+         */
         halted = true;
+        tick();
+        /*
+         * Mark running after #tick so #processAsyncTasks gets scheduled
+         */
+        running = false;
+        // Process async tasks and shutdown the asyncExecutor
+        asyncExecutor.shutdown();
```

Comment:
We should set an awaitTermination. We need it for 2 reasons: 1. Things shouldn't be able to run forever and 2. It's needed to process tasks still in queue

```
     * Initiates an orderly shutdown in which previously submitted
     * tasks are executed, but no new tasks will be accepted.
     * Invocation has no additional effect if already shut down.
     *
     * <p>This method does not wait for previously submitted tasks to
     * complete execution.  Use {@link #awaitTermination awaitTermination}
     * to do that.
```

Reply:
I was thinking that too, but wasn't sure what to have. Should we have it as 1 minute?

Reply:
Yeah a minute should be fine
```suggestion
        asyncExecutor.shutdown();
        asyncExecutor.awaitTermination(1, TimeUnit.MINUTES);
```